### PR TITLE
Fix #7268 - getComputedStyle should take `Element`, not `HTMLElement`

### LINKED ISSUE #7268: `getComputedStyle` should take `Element`, not `HTMLElement`.
Code:
 - `components/script/dom/webidls/Window.webidl` (use the commented-out line, keep the `[NewObject]` annotation)
 - `components/script/dom/window.rs` (update the `GetComputedStyle` implementation)
 - `components/script/dom/cssstyledeclaration.rs` (change the `owner` field to `JS<Element>` and fallout, including removing `PrivateCSSStyleDeclarationHelpers`)

I suspect there's no test yet. You'll have to write one that creates a non-HTML element (such as with `document.createElementNS("http://www.example.com/", "a")`), pass it to `window.getComputedStyle()` and check that it works. Add the file under `tests/wpt/mozilla/tests/css/`, and then run `./mach test-wpt _mozilla/css --manifest-update` to update the manifest. There's documentation at <http://testthewebforward.org/docs/> to help with writing the test.

### PR DESCRIPTION
**Pull request by mdibaiee**
This is my first patch, I hope I'm doing it right.

About the test, do you think this is enough and reliable?

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7288)

### CODE REVIEW

**Comment by Ms2ger** [comment]
Thank you for your PR. I have just a few comments below.

-S-awaiting-review +S-needs-code-changes 

---

Reviewed 4 of 8 files at r1, 1 of 1 files at r2.
Review status: 5 of 8 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

<sup>**[components/script/dom/htmlelement.rs, line 139 \[r2\]](https://reviewable.io:443/reviews/servo/servo/7288#-Jx9Cws4S40_7LANyGdX)** ([raw file](https://github.com/servo/servo/blob/75b9c7ca1f312c248f20b22b7299f052977b7337/components/script/dom/htmlelement.rs#L139)):</sup>
We prefer `ElementCast::from_ref`.

---

<sup>**[components/script/dom/webidls/Window.webidl, line 100 \[r2\]](https://reviewable.io:443/reviews/servo/servo/7288#-Jx9D85z731OwYmU8awM)** ([raw file](https://github.com/servo/servo/blob/75b9c7ca1f312c248f20b22b7299f052977b7337/components/script/dom/webidls/Window.webidl#L100)):</sup>
Remove this line entirely.

---

<sup>**[tests/wpt/mozilla/tests/css/computedstyle_on_element.html, line 13 \[r2\]](https://reviewable.io:443/reviews/servo/servo/7288#-Jx9DHxeVkHCHv6B7TNG)** ([raw file](https://github.com/servo/servo/blob/75b9c7ca1f312c248f20b22b7299f052977b7337/tests/wpt/mozilla/tests/css/computedstyle_on_element.html#L13)):</sup>
It's probably easier to make this a scripted test rather than a reftest. Please wrap all your code in `test(function() { ... })` and make this `assert_equals(style.opacity, "1");`

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7288)

**Comment by Ms2ger** [comment]
Reviewed 1 of 8 files at r1, 5 of 5 files at r3.
Review status: 6 of 7 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

<sup>**[tests/wpt/mozilla/tests/mozilla/element_getcomputedstyle.html, line 1 \[r3\]](https://reviewable.io:443/reviews/servo/servo/7288#-Jx9vNul_qINKH7aMnLM)** ([raw file](https://github.com/servo/servo/blob/2ac7dd7d860b082a9d08d1f74796b7b6b38ad04a/tests/wpt/mozilla/tests/mozilla/element_getcomputedstyle.html#L1)):</sup>
You're missing
```html
<script src="/resources/testharness.js"></script>
<script src="/resources/testharnessreport.js"></script>
```

(You'll need to regenerate the manifest again.)

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7288)

**Comment by Ms2ger** [comment]
Looking good! All you'll need to do now is squash the four commits into one, and update the commit message to include "fixes #7268" (literally, so github picks it up when we merge).

-S-awaiting-review +S-needs-squash 

---

Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7288)

**Comment by mdibaiee** [comment]
I'm not sure how it works, but it seems squashing commits already pushed doesn't work well, do I have to create another pull request with only one commit?

**Comment by Ms2ger** [comment]
You can delete your `computedstyle-element` branch locally, `git branch -m computedstyle computedstyle-element` and then `git push -f <your remote> computedstyle-element`

**Comment by mdibaiee** [comment]
@Ms2ger Oh right, done.

File: components/script/dom/cssstyledeclaration.rs [hunk 1]
```diff
@@ -10,7 +10,7 @@ use dom::bindings::js::{JS, Root};
 use dom::bindings::utils::{Reflector, reflect_dom_object};
 use dom::document::DocumentHelpers;
 use dom::element::{ElementHelpers, StylePriority};
-use dom::htmlelement::HTMLElement;
+use dom::element::Element;
```
Comment (Ms2ger) [inline components/script/dom/cssstyledeclaration.rs]: I'm afraid we introduced a new lint about sorting `use` statements that decided to bite you; sorry about that. Merging those two `use` lines should make it happy.

**Comment by mdibaiee** [comment]
@Ms2ger: Right, that's fixed now! Thanks for reviewing my PR.

**Comment by Ms2ger** [comment]
Thank you!

@bors-servo r+

**Comment by Ms2ger** [comment]
@bors-servo retry
* #6734

**Comment by jdm** [comment]
@bors-servo: retry
* #6734

**Pull request merged**